### DISCUSSION TITLE
multiple pipelines

### DISCUSSION THREAD

**Discussion by indra098124** [Alpine-DAV/ascent#1274] [Q&A]
Is it possible to define a pipeline on an existing pipeline? For e.g. I want to show a slice with vorticity contour but want to show only the region where x_vel > certain value? The current framework allows that?
Thank you

**Comment by nicolemarsaglia** [comment]
Hey @indra098124 it is possible! Here is a test we have for pipelines to pipelines that shows how the second pipeline (pl2) uses the first pipeline (pl1): https://github.com/Alpine-DAV/ascent/blob/06a09c3ebdbb4711c5fa5e3b7f345adaa26b96f8/src/tests/ascent/t_ascent_pipelines_to_pipelines.cpp#L80

For the case you describe, it sounds like you want to use a threshold filter on x_vel, then use the contour filter and slice filter. This can all be done in one pipeline, or you it can be done in multiple pipelines, depending on what you want to do. 

For example, the first pipeline could be the threshold and contour filters, and then you could have separate pipelines for individual slices. 

Let us know if you have any other questions!

**Reply by indra098124** [reply]
@nicolemarsaglia I understand that one pipeline operates on another. But, is it also possible to have parallel pipelines. For e.g. let's say I want to show two slices at different angles. For one slice I can use slice by plane, but how do I show the other slice?

**Reply by cyrush** [reply]
@indra098124  you can create as many pipelines as you like and in each scene as many plots as you like.

All the plots in the scene are drawn together, and each plot can select a different pipeline result to render

Here is a notional way to think about setting up two slices:
 
pipeline1 -> first slice
pipeline2 -> second slice

scene:
  plot1: pseudocolor of pipeline1
  plot2: pseudocolor of pipeline2

**Comment by indra098124** [comment]
@nicolemarsaglia thank you!
